TezosKit/Client/TezosNodeClient.swift

```diff
@@ -86,16 +86,22 @@ public class TezosNodeClient: AbstractClient {
   /// The default node URL to use.
   public static let defaultNodeURL = URL(string: "https://rpc.tezrpc.me")!
 
+  /// An factory which produces operations.
+  public let operationFactory: OperationFactory
+
   /// Initialize a new TezosNodeClient.
   /// - Parameters:
   ///   - remoteNodeURL: The path to the remote node, defaults to the default URL
+  ///   - tezosProtocol: The protocol version to use, defaults to athens.
   ///   - urlSession: The URLSession that will manage network requests, defaults to the shared session.
   ///   - callbackQueue: A dispatch queue that callbacks will be made on, defaults to the main queue.
   public init(
     remoteNodeURL: URL = defaultNodeURL,
+    tezosProtocol: TezosProtocol = .athens,
     urlSession: URLSession = URLSession.shared,
     callbackQueue: DispatchQueue = DispatchQueue.main
   ) {
+    operationFactory = OperationFactory(tezosProtocol: tezosProtocol)
```

Comment:
The way `tezosProtocol` is used, it seems to be a procol index, might be worth while specifying that. Protocol indices between mainnet and alphanet do not match. Testing new protocols with this library *may* require a recompile. Private networks may run totally different protocols.

Reply:
Good idea, I'll tack this added documentation into an upstream PR.